zebra-chain/src/sapling/keys.rs

```diff
@@ -841,12 +867,13 @@ impl TryFrom<[u8; 32]> for TransmissionKey {
     type Error = &'static str;
 
     /// Attempts to interpret a byte representation of an affine Jubjub point, failing if the
-    /// element is not on the curve, non-prime, the identity, or non-canonical.
+    /// element is not on the curve, non-canonical, or not in the prime-order subgroup.
     ///
     /// https://github.com/zkcrypto/jubjub/blob/master/src/lib.rs#L411
+    /// https://zips.z.cash/zip-0216
     fn try_from(bytes: [u8; 32]) -> Result<Self, Self::Error> {
         let affine_point = jubjub::AffinePoint::from_bytes(bytes).unwrap();
-        if affine_point.is_prime_order().into() {
+        if (affine_point.is_identity() | affine_point.is_prime_order()).into() {
```

Comment:
Can also use [`AffinePoint::is_torsion_free()`](https://github.com/zkcrypto/jubjub/blob/00ca002f6261f9a037a5e2ea5e3e60af53c88540/src/lib.rs#L441) if desired, but I also kinda like the explicit ' `is_identity() OR is_prime_order()`

Reply:
Thanks! I went with `is_torsion_free()` since it's more efficient, but added a comment explaining what it does